src/components/basicComponents/BaseButton.vue

```diff
@@ -1,10 +1,10 @@
 <template>
-  <button v-if="!link" :class="mode">
-    <slot></slot>
+  <button v-if="!link" :class="mode" value="value">
+    {{ text }}
   </button>
   <router-link v-else :to="to" :class="mode">
     <!-- router-link or router-view? -->
-    <slot></slot>
+    {{ text }}
```

Comment:
it depends what you need. If you need an icon here the slot will be better, but I don't think you'll use anything else than text :) 
I'd use v-html in a button: https://www.geeksforgeeks.org/vue-js-v-html-directive/

Reply:
Write me please what do you mean here, because I am not sure what is your idea of usage v-html here.